docs(serviceprovider): document deletion blocking behaviour (#129)
openmcp-project/docs: docs/developers/serviceprovider/02-develop.mdx

}
```

### Block deletion while user resources exist

> A naive `Delete` implementation removes the domain CRDs from the `ControlPlane` as soon as the user requests deletion of the service via the Onboarding cluster. Services such as Flux, Crossplane, or ESO install CRDs (e.g. `GitRepository`, `Bucket`, `ExternalSecret`) into the `ControlPlane`, and end users typically create custom resources of those CRDs. Removing the CRDs while user instances still exist orphans those resources in the cluster and can leave external systems (object storage, git repositories, secrets in vaults) escaping platform control.

Reviewer: Since this depends on the domain controller (e.g. Crossplane) deletion behavior, I suggest to slightly rephrase this.

```suggestion
A naive `Delete` implementation removes the domain CRDs from the `ControlPlane` as soon as the user requests deletion of the service via the Onboarding cluster. Services such as Flux, Crossplane, or ESO install CRDs (e.g. `GitRepository`, `Bucket`, `ExternalSecret`) into the `ControlPlane`, and end users typically create custom resources of those CRDs. Removing the CRDs while user instances still exist may orphan those resources in the cluster and leave external systems (object storage, git repositories, secrets in vaults) outside platform control.
```

Author: Fixed with [`56c63d2` (this PR)](https://github.com/openmcp-project/docs/pull/129/commits/56c63d289dcc84d5fdd61027b18dd0bd9d6e014a)

> Quality criterion [Deletion behaviour](./quality-criteria#1-deletion-behaviour) requires the service provider to block deletion in this case. The expected flow inside `Delete`:

Reviewer: I suggest to slightly rephrase this to put the focus on the idea that a service provider may improve the user experience by blocking deletion to prevent user data loss when a domain controller doesn't implement deletion safeguards (via CR finalizers).

Reviewer: Additionally it would be nice to address/distinguish between the two different clusters where a (end) users create objects in this context:

- onboarding cluster ([serviceprovider.API object](https://github.com/openmcp-project/opencontrolplane-runtime/blob/main/pkg/serviceprovider/reconciler.go#L22)) - controlled by the service provider
- (tenant) ManagedControlPlane cluster (domain CR objects) - controlled by the domain controllers managed by a service provider -> this is where a service provider developer can add the extra layer of protection to not  request deletion for a CRD until no CRs exist (basically forcing a user to clean up in cases where the domain controller doesn't care by not setting finalizers, see previous comment)

Author: Very good point! I have addressed your feedback via [`ea0b2aa` (this PR)](https://github.com/openmcp-project/docs/pull/129/commits/ea0b2aa2f9e778e0115611f153974f126b2a1798). What do you think @christophrj?
